File: src/main/java/com/demonwav/mcdev/update/ConfigurePluginUpdatesForm.java
package com.demonwav.mcdev.update;

import com.intellij.util.ui.AsyncProcessIcon;

import javax.swing.JButton;
import javax.swing.JComboBox;
import javax.swing.JLabel;
import javax.swing.JPanel;

public class ConfigurePluginUpdatesForm {
    private JComboBox<String> channelBox;
    private JButton checkForUpdatesNowButton;
    private JPanel panel;
    private AsyncProcessIcon updateCheckInProgressIcon;
    private JLabel updateStatusLabel;
    private JButton installButton;

    public JPanel getPanel() {
        return panel;
    }

    public JComboBox getChannelBox() {
        return channelBox;
    }

    public JButton getCheckForUpdatesNowButton() {
        return checkForUpdatesNowButton;
    }

    public AsyncProcessIcon getUpdateCheckInProgressIcon() {
        return updateCheckInProgressIcon;
    }

    public JLabel getUpdateStatusLabel() {
        return updateStatusLabel;
    }

    public JButton getInstallButton() {
        return installButton;
    }

    private void createUIComponents() {
        updateCheckInProgressIcon = new AsyncProcessIcon("Plugin update check in progress");
    }
}

File: src/main/kotlin/com/demonwav/mcdev/update/ConfigurePluginUpdatesForm.kt
/*
 * Minecraft Dev for IntelliJ
 *
 * https://minecraftdev.org
 *
 * Copyright (c) 2019 minecraft-dev
 *
 * MIT License
 */

package com.demonwav.mcdev.update

import com.intellij.util.ui.AsyncProcessIcon
import javax.swing.JButton
import javax.swing.JComboBox
import javax.swing.JLabel
import javax.swing.JPanel

class ConfigurePluginUpdatesForm {
    lateinit var channelBox: JComboBox<String>
    lateinit var checkForUpdatesNowButton: JButton
    lateinit var panel: JPanel
    lateinit var updateCheckInProgressIcon: AsyncProcessIcon

    lateinit var updateStatusLabel: JLabel
    lateinit var installButton: JButton

    private fun createUIComponents() {
        updateCheckInProgressIcon = AsyncProcessIcon("Plugin update check in progress")
    }
}
